packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefModifyBasis.hpp

```diff
@@ -151,121 +291,40 @@ namespace Intrepid2 {
-          orts(cell).getEdgeOrientation(ortEdges, numEdges);
-          
-          // apply coeff matrix
-          for (ordinal_type edgeId=0;edgeId<numEdges;++edgeId) {
-            const ordinal_type ordEdge = (1 < tagToOrdinal.extent(0) ? (static_cast<size_type>(edgeId) < tagToOrdinal.extent(1) ? tagToOrdinal(1, edgeId, 0) : -1) : -1);
-            
-            if (ordEdge != -1) {
-              existEdgeDofs = 1;
-              const ordinal_type ndofEdge = ordinalToTag(ordEdge, 3);
-              const auto mat = Kokkos::subview(matData, 
-                                               edgeId, ortEdges[edgeId], 
-                                               Kokkos::ALL(), Kokkos::ALL());
-              
-              for (ordinal_type j=0;j<numPoints;++j) 
-                for (ordinal_type i=0;i<ndofEdge;++i) {
-                  const ordinal_type ii = tagToOrdinal(1, edgeId, i);
-                  
-                  for (ordinal_type k=0;k<dimBasis;++k) {
-                    input_value_type temp = 0.0;
-                    for (ordinal_type l=0;l<ndofEdge;++l) {
-                      const ordinal_type ll = tagToOrdinal(1, edgeId, l);
-                      temp += mat(i,l)*in(ll, j, k);
-                    }
-                    out(ii, j, k) = temp;
-                  }
-                }
-            }
-          }
-        }
-        
-        // face transformation
-        if (numFaces > 0) {
-          ordinal_type ortFaces[12];
-          orts(cell).getFaceOrientation(ortFaces, numFaces);
-          
-          // apply coeff matrix
-          for (ordinal_type faceId=0;faceId<numFaces;++faceId) {
-            const ordinal_type ordFace = (2 < tagToOrdinal.extent(0) ? (static_cast<size_type>(faceId) < tagToOrdinal.extent(1) ? tagToOrdinal(2, faceId, 0) : -1) : -1);
-            
-            if (ordFace != -1) {
-              const ordinal_type ndofFace = ordinalToTag(ordFace, 3);
-              const auto mat = Kokkos::subview(matData, 
-                                               numEdges*existEdgeDofs+faceId, ortFaces[faceId], 
-                                               Kokkos::ALL(), Kokkos::ALL());
-              
-              for (ordinal_type j=0;j<numPoints;++j) 
-                for (ordinal_type i=0;i<ndofFace;++i) {
-                  const ordinal_type ii = tagToOrdinal(2, faceId, i);
-                  
-                  for (ordinal_type k=0;k<dimBasis;++k) {
-                    input_value_type temp = 0.0;
-                    for (ordinal_type l=0;l<ndofFace;++l) {
-                      const ordinal_type ll = tagToOrdinal(2, faceId, l);
-                      temp += mat(i,l)*in(ll, j, k);
-                    }
-                    out(ii, j, k) = temp;
-                  }
-                }
-            }
-          }
-        }
-        
-      }
+
+      const Kokkos::RangePolicy<SpT> policy(0, numCells);
+      typedef F_modifyBasisByOrientation
+        <decltype(orts),
```

Comment:
Best practice with the "implementation struct and templated function" pattern using Kokkos::View is to have the function "canonicalize" the struct's template arguments.  This avoids redundant specializations of the struct.

Reply:
I originally want to use labmda but somehow it does not compile. I just make it as a functor. Here look and feel does not really matter. Thanks if you can approve the changes.

Reply:
Done! :-)